rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -58,14 +65,33 @@ def __init__(
             base_tools=base_tools,
             msg_aggregator=msg_aggregator,
         )
+        self.vaults: dict[str, set[ChecksumEvmAddress]] = {
+            YEARN_VAULTS_V1_PROTOCOL: set(),
+            YEARN_VAULTS_V2_PROTOCOL: set(),
+        }
+
+    def reload_data(self) -> None:
+        """Check that cache is up to date and refresh cache from db"""
+        if should_update_protocol_cache(CacheType.YEARN_VAULTS) is True:
+            query_yearn_vaults(db=self.evm_inquirer.database)
+        elif len(self.vaults[YEARN_VAULTS_V2_PROTOCOL]) != 0:
+            return  # we didn't update the globaldb cache and we have the data already
 
         with GlobalDBHandler().conn.read_ctx() as cursor:
-            db_query = 'SELECT address FROM evm_tokens WHERE protocol=? AND chain=?'
-            cursor.execute(db_query, (YEARN_VAULTS_V1_PROTOCOL, ChainID.ETHEREUM.serialize_for_db()))  # noqa: E501
-            self.vaults_v1 = {string_to_evm_address(row[0]) for row in cursor}
-
-            cursor.execute(db_query, (YEARN_VAULTS_V2_PROTOCOL, ChainID.ETHEREUM.serialize_for_db()))  # noqa: E501
-            self.vaults_v2 = {string_to_evm_address(row[0]) for row in cursor}
+            bindings = (
+                YEARN_VAULTS_V1_PROTOCOL,
+                YEARN_VAULTS_V2_PROTOCOL,
+                ChainID.ETHEREUM.serialize_for_db(),
+            )
+            query_body = 'FROM evm_tokens WHERE protocol IN (?, ?) AND chain=?'
```

Comment:
first query, then bindings. This is how it's all over the code

```suggestion
           query_body = 'FROM evm_tokens WHERE protocol IN (?, ?) AND chain=?'
            bindings = (
                YEARN_VAULTS_V1_PROTOCOL,
                YEARN_VAULTS_V2_PROTOCOL,
                ChainID.ETHEREUM.serialize_for_db(),
            )
```